blog/2025-01-21-opencost-mcp-server/index.mdx

```diff
@@ -0,0 +1,104 @@
+---
+slug: opencost-mcp-server
+title: "OpenCost MCP Server: AI-Powered Cost Management for Kubernetes"
+authors: [adeshpal]
+tags: [MCP, AI, cost management, Kubernetes, automation]
+---
+
+![OpenCost MCP Server](./img/opencost-mcp-server.png)
+
+The future of Kubernetes cost management is here, and it's powered by AI. We're excited to announce that OpenCost now includes a built-in **Model Context Protocol (MCP) server** that enables AI agents to directly access and analyze your Kubernetes cost data. This groundbreaking integration brings intelligent automation to cost optimization, making it easier than ever to understand, predict, and optimize your cloud spending.
+
+<!--truncate-->
+
+## What is Model Context Protocol (MCP)?
+
+The Model Context Protocol is an open standard introduced by Anthropic in late 2024 that standardizes how AI systems integrate with external tools and data sources. Think of MCP as a universal translator that allows AI agents to seamlessly interact with your infrastructure, databases, and applications.
+
+![MCP Architecture](./img/mcp-workflow.png)
+
+For OpenCost, this means AI agents can now:
+- **Query cost data** in real-time using natural language
+- **Analyze spending patterns** across namespaces, pods, and nodes
+- **Generate cost reports** and recommendations automatically
+- **Monitor budget alerts** and cost anomalies
+- **Optimize resource allocation** based on historical data
+
+## OpenCost MCP Server: Zero-Configuration AI Integration
+
+The best part? OpenCost's MCP server is **enabled by default** in every deployment. No additional setup, no complex configuration—just intelligent cost management out of the box.
+
+### Key Features
+
+- **🚀 Enabled by Default**: MCP server starts automatically with OpenCost
+- **🔧 Full User Control**: Easy to disable or configure ports and settings
+- **📊 Comprehensive Data Access**: Query allocation, asset, and cloud cost data
+- **🌐 HTTP Transport**: Reliable communication with MCP clients
+- **☸️ Helm Integration**: Built into the official Helm chart for production deployments
+- **🔍 Advanced Filtering**: Filter by provider, service, region, and more
```

Comment:
just say that this supports all the filters and features of the battle tested APIs